base/loading.jl

```diff
@@ -145,7 +145,7 @@ function show(io::IO, ex::PrecompilableError)
     end
 end
 precompilableerror(ex::PrecompilableError, c) = ex.isprecompilable == c
-precompilableerror(ex::LoadError, c) = precompilableerror(ex.error, c)
+precompilableerror(ex::Union{LoadError,InitError}, c) = precompilableerror(ex.error, c)
 precompilableerror(ex, c) = false
```

Comment:
I had thought about it, but it did seem like a pretty small corner case. I suppose there's no harm in future-proofing though.